### DISCUSSION TITLE
can't covert anndata to seurat

### DISCUSSION THREAD

**Discussion by cdadams-harvard** [satijalab/seurat#5907] [Q&A] [answered]
Hello, I am trying to convert a GTEx snRNA-seq file stored as AnnaData to seurat, but am having trouble. 

This is the file: 
https://storage.googleapis.com/gtex_analysis_v9/snrna_seq_data/GTEx_8_tissues_snRNAseq_immune_atlas_071421.public_obs.h5ad

I have loaded the following libraries: 
library(dplyr)
library(Seurat)
library(patchwork)
library(SeuratData)
library(SeuratDisk)
library(anndata)
library(reticulate)
library(Matrix)
library(gdata)

I have tried the "Convert" command: 

Convert("GTEx_8_tissues_snRNAseq_atlas_071421.public_obs.h5ad", dest = "h5seurat", overwrite = TRUE)
gtex <- LoadH5Seurat("GTEx_8_tissues_snRNAseq_atlas_071421.public_obs.h5ad.h5seurat")

I get this error: 

Validating h5Seurat file
Error in h5attr(x = self[["reductions"]][[reduc]], which = "active.assay") : 
  Attribute does not exist

I have tried loading it into R first: 

gtex <- ad$read_h5ad("GTEx_8_tissues_snRNAseq_atlas_071421.public_obs.h5ad")
gtex2 <- Convert(gtex, to = "seurat")

This produces another error:

Error in UseMethod(generic = "Convert", object = source) : 
  no applicable method for 'Convert' applied to an object of class "c('anndata._core.anndata.AnnData', 'python.builtin.object')"

Is there another way or a fix? 

Thanks, 
Charleen

**Comment by denvercal1234GitHub** [comment]
https://github.com/denvercal1234GitHub/anndata_Seurat_Utilities

**Comment by samuel-marsh** [comment, marked as answer]
Th scverse team has released an official R package to perform such conversions.
https://github.com/scverse/anndataR
